java/apply-scheduled-updates-to-preplanned-map-area/src/main/java/com/esri/arcgisruntime/sample/applyscheduledupdatestopreplannedmaparea/MainActivity.java

```diff
@@ -0,0 +1,305 @@
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.tasks.geodatabase.SyncGeodatabaseParameters;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapSyncJob;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapSyncParameters;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapSyncResult;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapSyncTask;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapUpdatesInfo;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineUpdateAvailability;
+import com.esri.arcgisruntime.tasks.offlinemap.PreplannedScheduledUpdatesOption;
+
+public class MainActivity extends AppCompatActivity {
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+
+  private MapView mMapView;
+  private File mCopyOfMmpk;
+  private TextView mUpdateAvailableTextView;
+  private TextView mUpdateSizeTextView;
+  private Button mApplyScheduledUpdatesButton;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // create a map view
+    mMapView = findViewById(R.id.mapView);
+
+    // get a reference to the UI views
+    mUpdateAvailableTextView = findViewById(R.id.updateAvailableTextView);
+    mUpdateSizeTextView = findViewById(R.id.updateSizeTextView);
+    mApplyScheduledUpdatesButton = findViewById(R.id.applyScheduledUpdatesButton);
+
+    // this is the original mmpk, not updated by the scheduled update
+    File originalMmpk = new File(Environment.getExternalStorageDirectory() + getString(R.string.canyonlands_mmpk_path));
+    // copy of the mmpk file which will have the update applied to it
+    mCopyOfMmpk = new File(getCacheDir() + getString(R.string.canyonlands_folder));
+
+    try {
+      // copy the original mmpk into the cache, overwriting any copy of the mmpk already there
+      copyDirectory(originalMmpk, mCopyOfMmpk);
+    } catch (IOException e) {
+      Log.e(TAG, "Error copying MMPK file: " + e.getMessage());
+    }
+
+    requestReadPermission();
+  }
+
+  private void applyScheduledUpdate() {
+
+    // load the offline map as a mobile map package
+    MobileMapPackage mobileMapPackage = new MobileMapPackage(mCopyOfMmpk.getPath());
+    mobileMapPackage.loadAsync();
+    mobileMapPackage.addDoneLoadingListener(() -> {
+      if (mobileMapPackage.getLoadStatus() == LoadStatus.LOADED && !mobileMapPackage.getMaps().isEmpty()) {
+        // add the map from the mobile map package to the map view
+        ArcGISMap offlineMap = mobileMapPackage.getMaps().get(0);
+        mMapView.setMap(offlineMap);
+        // create an offline map sync task with the preplanned area
+        OfflineMapSyncTask offlineMapSyncTask = new OfflineMapSyncTask(offlineMap);
+        // check for updates to the offline map
+        ListenableFuture<OfflineMapUpdatesInfo> offlineMapUpdatesInfoFuture = checkForUpdatesFuture(offlineMapSyncTask);
```

Comment:
This is looking good - I've done it slightly in differently in Java. The checkForUpdates() method doesn't return anything, but only enables the button to trigger applying updates if there are any available. Something along those lines:
```
checkForUpdate(){
    if (update available){
     update UI to show update
     enableUpdateButton()
    } else {
     update UI to show no update
    }    
}
resolveApplyUpdateButtonPress(){
    runTheUpdateTask()
    checkForUpdate()
}
```
It might reduce the amount of nesting a little, but really comes down to preference.

Reply:
This is a good point. I think I'm going to refactor it into something that just handles the UI....

Reply:
It's cool to see a method that returns a `ListenableFuture` though!